mindmeld/components/question_answerer.py

```diff
@@ -451,803 +528,2178 @@ def _generate_mapping_data(embedder_model, embedding_fields):
+            else:
+                raise ValueError("query_type in processor must contain `text` or `keyword` string")
+
+            if isinstance(string_or_strings, str):
+                return processor(string_or_strings)
+            elif isinstance(string_or_strings, (list, set)):
+                return [processor(string) for string in string_or_strings]
+            else:
+                raise ValueError("Input to auto processor must be string or list of strings")
+
+        @staticmethod
+        def is_bool(value):
+            return isinstance(value, bool)
+
+        @staticmethod
+        def is_number(value):
+            return isinstance(value, numbers.Number)
+
+        @staticmethod
+        def is_string(value):
+            return isinstance(value, str)
+
+        @staticmethod
+        def is_list_of_strings(value):
+            return (isinstance(value, (list, set))
+                    and len(value) > 0
+                    and all([isinstance(val, str) for val in value]))
+
+        @staticmethod
+        def is_date(value):
+            for fmt in NonElasticsearchQuestionAnswerer.FieldResource.DATE_FORMATS:
+                try:
+                    datetime.datetime.strptime(value, fmt)
+                    return True
+                except (ValueError, TypeError):
+                    pass
+            return False
+
+        @staticmethod
+        def is_location(value):
+            return (
+                (isinstance(value, dict) and "lat" in value and "lon" in value)
+                or (isinstance(value, list) and len(value) == 2 and
+                    isinstance(value[0], numbers.Number) and isinstance(value[1], numbers.Number))
+                or (isinstance(value, str) and "," in value and len(value.split(",")) == 2)
+            )
+
+        @staticmethod
+        def number_scorer(some_number):
+            return some_number
+
+        @staticmethod
+        def date_scorer(some_date, origin_date=datetime.datetime.now()):
+            """
+            returns # of days as score
+            """
+            target_date = origin_date
+            for fmt in NonElasticsearchQuestionAnswerer.FieldResource.DATE_FORMATS:
+                try:
+                    target_date = datetime.datetime.strptime(some_date, fmt)
+                    break
+                except ValueError:
+                    pass
+            return (target_date - origin_date).days
+
+        @staticmethod
+        def location_scorer(some_location, origin_location):
+            # assumes both args are of format "37.77,122.41" where comma seperated lat and lon
+
+            R = 6373.0
+            lat1, lon1 = [float(ii) for ii in some_location.split(",")]
+            lat2, lon2 = [float(ii) for ii in origin_location.split(",")]
+
+            dlon = radians(lon2) - radians(lon1)
+            dlat = radians(lat2) - radians(lat1)
+
+            a = sin(dlat / 2) ** 2 + cos(lat1) * cos(lat2) * sin(dlon / 2) ** 2
```

Comment:
This is a bit hard to understand, documentation here would be good.

Reply:
Added!